From 1991e3d446a111a873c00835687526b1c3620197 Mon Sep 17 00:00:00 2001
From: Marc Lainez <marc.lainez@gmail.com>
Date: Fri, 13 Mar 2026 09:04:42 +0100
Subject: [PATCH 06/11] arm64: dts: qcom: pmi632: add QGauge fuel gauge node

Add the QGauge (QG) fuel gauge device node at register offset 0x4800
in the PMI632 PMIC. The node defines all 7 hardware interrupts
(batt-missing, vbat-low, vbat-empty, fifo-done, good-ocv,
fsm-state-chg, event) and is disabled by default, to be enabled
by board-level DTS files that provide a monitored-battery reference.

Signed-off-by: Marc Lainez <marc.lainez@gmail.com>
---
 arch/arm64/boot/dts/qcom/pmi632.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/pmi632.dtsi b/arch/arm64/boot/dts/qcom/pmi632.dtsi
index 8c899d148..4a3f0bf9f 100644
--- a/arch/arm64/boot/dts/qcom/pmi632.dtsi
+++ b/arch/arm64/boot/dts/qcom/pmi632.dtsi
@@ -156,6 +156,26 @@ pmi632_adc_tm: adc-tm@3500 {
 			status = "disabled";
 		};
 
+		pmi632_qg: fuel-gauge@4800 {
+			compatible = "qcom,pmi632-qg";
+			reg = <0x4800>;
+			interrupts = <0x2 0x48 0x0 IRQ_TYPE_EDGE_BOTH>,
+				     <0x2 0x48 0x1 IRQ_TYPE_EDGE_BOTH>,
+				     <0x2 0x48 0x2 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x48 0x3 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x48 0x4 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x48 0x5 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x48 0x6 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "batt-missing",
+					  "vbat-low",
+					  "vbat-empty",
+					  "fifo-done",
+					  "good-ocv",
+					  "fsm-state-chg",
+					  "event";
+			status = "disabled";
+		};
+
 		pmi632_pbs_client3: pbs@7400 {
 			compatible = "qcom,pmi632-pbs", "qcom,pbs";
 			reg = <0x7400>;
-- 
2.43.0

